packages/nx-monorepo/src/nx-monorepo.ts

```diff
@@ -491,12 +486,8 @@ export class NxMonorepoProject extends TypeScriptProject {
    * @param dependent project you want to have the dependency.
    * @param dependee project you wish to depend on.
    */
-  public addImplicitDependency(dependent: Project, dependee: Project) {
-    if (this.implicitDependencies[dependent.name]) {
-      this.implicitDependencies[dependent.name].push(dependee.name);
-    } else {
-      this.implicitDependencies[dependent.name] = [dependee.name];
-    }
+  public addImplicitDependency(dependent: Project, dependee: Project | string) {
+    NxProject.of(dependent)?.addImplicitDependency(dependee);
```

Comment:
Should we throw error if `dependent` doesn't have `NxProject`? Otherwise it is a noop without warning that the implicit dependency wasn't set? I know we are adding NxProject to subProjects above, but to would be nice to future proof.

Reply:
Updated.